Support felt constant template parameters in witness generation (#604)
Veridise/llzk-lib: doc/doxygen/03_syntax.md

## Semantic Rules

- Each `array.new` operation creates a fresh mutable array allocation. Two identical `array.new` operations are not interchangeable when either result may be read or written. The same is true for `pod.new`.

> - Felt constants are valid template arguments for parameterized struct types, free-function calls, and `verif.include` contract inclusions. A fieldless constant adopts the required felt type; an explicitly fielded constant can materialize only when its type exactly matches the required felt type.

Reviewer: ```suggestion
- Felt constants are valid template arguments for parameterized struct types, free-function calls, and `verif.include` contract inclusions. A fieldless constant adopts the required felt type; a constant that specifies a field can materialize only when its type exactly matches the required felt type.
```
